packages/sdk/src/across/rateModel.ts

```diff
@@ -0,0 +1,143 @@
+import { expectedRateModelKeys, RateModel } from "./constants";
+import { exists } from "../utils";
+import { ethers } from "ethers";
+
+// Each L1 token is mapped to an array of stringified rate models, ordered by the block height at which they were
+// published on-chain. This dictionary is used internally to fetch a rate model for a block height.
+type RateModelEventsDictionary = {
+  [l1TokenAddress: string]: { blockNumber: number; rateModel: string }[];
+};
+
+// Events should be reformatted into this shape to be used as input into methods in this file.
+export type RateModelEvent = {
+  blockNumber: number;
+  rateModel: string;
+  l1Token: string;
+};
+
+/**
+ * Helper method that returns parsed rate model from string, or throws.
+ * @param rateModelString Stringified rate model to parse.
+ * @returns Rate model object. Must conform to `expectedRateModelKeys` format.
+ */
+
+export const parseAndReturnRateModelFromString = (rateModelString: string): RateModel => {
+  const rateModelFromEvent = JSON.parse(rateModelString);
+
+  // Rate model must contain all keys in `expectedRateModelKeys`, and extra keys are OK.
+  for (const key in expectedRateModelKeys) {
+    if (!(key in Object.keys(rateModelFromEvent))) {
```

Comment:
nit: I don't think this does what you might expect. `in` checks that the `key` is a _key_ for the other object. In this case that means, it's checking that `key` is 1, 2, 3, or 4. The `in` operator is a bit confusing because of how it deals with arrays.

I think to make this do what we want, we'd want to change this line to
```ts
if (!(key in rateModelFromEvent)) {
```

Reply:
I don't think this works either assuming my tests are correct so I've just replaced this with the much easier to reason about `!Object.keys(rateModelFromEvent).includes(key)`

Reply:
Weird. I was able to verify this behavior in the console:
```
> const obj = { a: 1, b: 2 }
undefined
> "a" in obj
true
> "a" in Object.keys(obj)
false
```

Reply:
Either way, `includes` is totally fine to!